x-pack/legacy/plugins/ml/server/routes/anomaly_detectors.ts

```diff
@@ -0,0 +1,460 @@
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.validateDetector', {
+          body: request.body,
+        });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {post} /api/ml/anomaly_detectors/:jobId/_forecast Create forecast for specified job
+   * @apiName ForecastAnomalyDetector
+   * @apiDescription Creates forecast for specified anomaly detection job.
+   *
+   * @apiParam {String} jobId Job ID.
+   */
+  router.post(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/_forecast',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+        body: schema.object({ duration: schema.any() }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const jobId = request.params.jobId;
+        const duration = request.body.duration;
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.forecast', {
+          jobId,
+          duration,
+        });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {post} /api/ml/anomaly_detectors/:jobId/results/overall_buckets  Obtain overall bucket scores for the specified job ID
+   * @apiName ForecastAnomalyDetector
+   * @apiDescription Creates forecast for specified anomaly detection job.
+   *
+   * @apiParam {String} jobId Job ID.
+   *
+   * @apiSuccess {Number} count
+   * @apiSuccess {Object[]} overall_buckets
+   */
+  router.post(
+    {
+      path: '/api/ml/anomaly_detectors/{jobId}/results/overall_buckets',
+      validate: {
+        params: schema.object({
+          jobId: schema.string(),
+        }),
+        body: schema.object({
+          topN: schema.number(),
+          bucketSpan: schema.string(),
+          start: schema.number(),
+          end: schema.number(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const results = await context.ml!.mlClient.callAsCurrentUser('ml.overallBuckets', {
+          jobId: request.params.jobId,
+          top_n: request.body.topN,
+          bucket_span: request.body.bucketSpan,
+          start: request.body.start,
+          end: request.body.end,
+        });
+        return response.ok({
+          body: { ...results },
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup AnomalyDetectors
+   *
+   * @api {get} /api/ml/anomaly_detectors/:jobId/results/categories/{categoryId} Get results category data by job id and category id
```

Comment:
typo,  has to be `:categoryId `
```suggestion
   * @api {get} /api/ml/anomaly_detectors/:jobId/results/categories/:categoryId Get results category data by job id and category id
```

Reply:
Thanks for catching that. Updated in b70059756cf4d0eb6a125b0ba7e891c69daa622d